src/main/kotlin/com/ampnet/userservice/service/impl/ReCaptchaServiceImpl.kt

```diff
@@ -0,0 +1,65 @@
+package com.ampnet.userservice.service.impl
+
+import com.ampnet.userservice.config.ApplicationProperties
+import com.ampnet.userservice.exception.ReCaptchaException
+import com.ampnet.userservice.service.ReCaptchaService
+import com.ampnet.userservice.service.pojo.GoogleResponse
+import com.ampnet.userservice.service.pojo.ReCaptchaRequest
+import com.fasterxml.jackson.core.JsonProcessingException
+import com.fasterxml.jackson.databind.ObjectMapper
+import org.springframework.http.ResponseEntity
+import org.springframework.stereotype.Service
+import org.springframework.web.client.RestClientException
+import org.springframework.web.client.RestTemplate
+import org.springframework.web.client.postForEntity
+
+@Service
+class ReCaptchaServiceImpl(
+    private val applicationProperties: ApplicationProperties,
+    private val restTemplate: RestTemplate,
+    private val objectMapper: ObjectMapper
+) : ReCaptchaService {
+
+    @Throws(ReCaptchaException::class)
+    override fun validateResponseToken(reCaptchaToken: String) {
+        if (!applicationProperties.reCaptcha.enabled) return
+        val request = ReCaptchaRequest(
+            applicationProperties.reCaptcha.secret,
+            reCaptchaToken
+        )
+        try {
+            val responseEntity = restTemplate.postForEntity<String>(
+                applicationProperties.reCaptcha.url, request
+            )
```

Comment:
The request is wrong! https://developers.google.com/recaptcha/docs/verify
Data should be sent as URL params.